### PR TITLE
data: scaladocs

### PR DESCRIPTION
**Pull request by fwbrasil**

### CODE REVIEW

File: kyo-data/shared/src/main/scala/kyo/Ansi.scala [hunk 1]
```diff
@@ -1,58 +1,112 @@
...
-                    if isHeader || isTrailer then
-                        line.green
-                    else
-                        highlightLine(line.drop(toDrop))
-
-                s"${lineNumber.dim} $separator $highlightedLine"
-            }
-
-            processedLines.mkString("\n")
+            try
```
Comment (fwbrasil) [inline kyo-data/shared/src/main/scala/kyo/Ansi.scala:69]: An unrelated improvement: this method is now used in several places to report errors. If it throws an exception, we'd lose visibility of the underlying failure so I've added a fallback to return the code without highlighting if that happens.

File: kyo-data/shared/src/main/scala/kyo/Result.scala [hunk 2]
```diff
@@ -7,21 +7,47 @@ import scala.reflect.ClassTag
...
+      * @tparam A
+      *   The type of the successful result
+      * @return
+      *   A Result containing either the successful value or a Panic with the caught exception
+      */
     inline def apply[A](inline expr: => A): Result[Nothing, A] =
         try
             Success(expr)
         catch
             case ex => Panic(ex)
```
Comment (hearnadam) [inline kyo-data/shared/src/main/scala/kyo/Result.scala:39]: `NonFatal`? Line 56 as well?

File: kyo-data/shared/src/main/scala/kyo/Result.scala [hunk 3]
```diff
@@ -32,10 +58,56 @@ object Result:
...
+      * @param value
+      *   The successful value
+      * @tparam E
+      *   The type of the error (not used in this case)
+      * @tparam A
+      *   The type of the successful value
+      * @return
+      *   A successful Result
+      */
+    inline def success[E, A](inline value: A): Result[E, A] = Success(value)
```
Comment (hearnadam) [inline kyo-data/shared/src/main/scala/kyo/Result.scala:72]: style: I think this should be renamed to `succeed` (perhaps I am too used to ZIO)

File: kyo-data/shared/src/main/scala/kyo/Result.scala [hunk 4]
```diff
@@ -220,24 +533,51 @@ object Result:
...
+        /** Swaps the success and failure cases of the Result.
+          *
+          * @return
+          *   A new Result with success and failure swapped
+          */
         def swap: Result[A, E] =
             self match
                 case Fail(e)    => Result.success(e)
                 case Success(v) => Result.fail(v)
                 case _          => self.asInstanceOf[Result[A, E]]
```
Comment (hearnadam) [inline kyo-data/shared/src/main/scala/kyo/Result.scala:574]: Should we have a toString impl of `SuccessError`? `"Success(" * (depth - 1) + ...`

**Review by hearnadam — APPROVED** [review]
LGTM - mostly pre-existing comments.

File: kyo-data/shared/src/main/scala/kyo/Result.scala (hunk 2 above)
Comment (fwbrasil) [reply kyo-data/shared/src/main/scala/kyo/Result.scala:39]: I put the `NonFatal` check in `Panic.apply` to reduce the size of the inlined bytecode.

Comment (hearnadam) [reply kyo-data/shared/src/main/scala/kyo/Result.scala:39]: Ah clever, I must have forgotten.

File: kyo-data/shared/src/main/scala/kyo/Result.scala (hunk 4 above)
Comment (fwbrasil) [reply kyo-data/shared/src/main/scala/kyo/Result.scala:574]: Makes sense, I've added a `toString` to `DefinedEmpty` in `Maybe` as well: https://github.com/getkyo/kyo/pull/639/commits/852c90755c1761588cb88c235477e50d18cd3623

File: kyo-data/shared/src/main/scala/kyo/Result.scala (hunk 3 above)
Comment (fwbrasil) [reply kyo-data/shared/src/main/scala/kyo/Result.scala:72]: let's follow up separately

File: kyo-data/shared/src/main/scala/kyo/Maybe.scala [hunk 5]
```diff
@@ -142,6 +403,9 @@ object Maybe:
                     Empty
             def nest =
                 DefinedEmpty(depth + 1)
+
+            override def toString: String =
+                "Maybe(" * depth + "Empty" + ")" * depth
```
Comment (hearnadam) [inline kyo-data/shared/src/main/scala/kyo/Maybe.scala:408]: I think this should be `Defined(`?

**Pull request merged**